packages/firebase_ml_vision/lib/src/barcode_detector.dart

```diff
@@ -4,18 +4,599 @@
+
+  /// Barcode format constant for Code 128.
+  static const VisionBarcodeFormat Code128 =
+      const VisionBarcodeFormat._(0x0001);
+
+  /// Barcode format constant for Code 39.
+  static const VisionBarcodeFormat Code39 = const VisionBarcodeFormat._(0x0002);
+
+  /// Barcode format constant for Code 93.
+  static const VisionBarcodeFormat Code93 = const VisionBarcodeFormat._(0x0004);
+
+  /// Barcode format constant for Codabar.
+  static const VisionBarcodeFormat CodaBar =
+      const VisionBarcodeFormat._(0x0008);
+
+  /// Barcode format constant for Data Matrix.
+  static const VisionBarcodeFormat DataMatrix =
+      const VisionBarcodeFormat._(0x0010);
+
+  /// Barcode format constant for EAN-13.
+  static const VisionBarcodeFormat EAN13 = const VisionBarcodeFormat._(0x0020);
+
+  /// Barcode format constant for EAN-8.
+  static const VisionBarcodeFormat EAN8 = const VisionBarcodeFormat._(0x0040);
+
+  /// Barcode format constant for ITF (Interleaved Two-of-Five).
+  static const VisionBarcodeFormat ITF = const VisionBarcodeFormat._(0x0080);
+
+  /// Barcode format constant for QR Code.
+  static const VisionBarcodeFormat QRCode = const VisionBarcodeFormat._(0x0100);
+
+  /// Barcode format constant for UPC-A.
+  static const VisionBarcodeFormat UPCA = const VisionBarcodeFormat._(0x0200);
+
+  /// Barcode format constant for UPC-E.
+  static const VisionBarcodeFormat UPCE = const VisionBarcodeFormat._(0x0400);
+
+  /// Barcode format constant for PDF-417.
+  static const VisionBarcodeFormat PDF417 = const VisionBarcodeFormat._(0x0800);
+
+  /// Barcode format constant for AZTEC.
+  static const VisionBarcodeFormat Aztec = const VisionBarcodeFormat._(0x1000);
+}
+
+/// Barcode value type constants - enumeration of supported barcode content value types
+enum VisionBarcodeValueType {
+  /// Unknown Barcode value types.
+  Unknown,
+
+  /// Barcode value type for contact info.
+  ContactInfo,
+
+  /// Barcode value type for email addresses.
+  Email,
+
+  /// Barcode value type for ISBNs.
+  ISBN,
+
+  /// Barcode value type for phone numbers.
+  Phone,
+
+  /// Barcode value type for product codes.
+  Product,
+
+  /// Barcode value type for SMS details.
+  SMS,
+
+  /// Barcode value type for plain text.
+  Text,
+
+  /// Barcode value type for URLs/bookmarks.
+  URL,
+
+  /// Barcode value type for Wi-Fi access point details.
+  WiFi,
+
+  /// Barcode value type for geographic coordinates.
+  GeographicCoordinates,
+
+  /// Barcode value type for calendar events.
+  CalendarEvent,
+
+  /// Barcode value type for driver's license data.
+  DriversLicense,
+}
+
+/// An email message from a 'MAILTO:' or similar QRCode type.
+class VisionBarcodeEmail {
+  VisionBarcodeEmail._(Map<dynamic, dynamic> data)
+      : type = VisionBarcodeEmailType.values.elementAt(data['type']),
+        address = data['address'],
+        body = data['body'],
+        subject = data['subject'];
+
+  /// Gets email's address.
```

Comment:
Remove `Gets`. Same for `body` and `subject`

Using `The email's address` would be a good choice.